src/graph.cc

```diff
@@ -28,7 +28,10 @@
 #include "util.h"
 
 bool Node::Stat(DiskInterface* disk_interface, string* err) {
-  return (mtime_ = disk_interface->Stat(path_, err)) != -1;
+  METRIC_RECORD("node stat");
+  mtime_ = disk_interface->Stat(path_, err);
+  exists_ = (mtime_ > 0) ? ExistanceStatusExists : ExistanceStatusMissing;
```

Comment:
wait - this will be incorrect now that the Flatpak-hack is gone, right?

Reply:
True. I've reverted that to the original `disk_interface.cc` code to have a branch that is working correctly. It is a too big change for me to refactor the interface of `DiskInterface::Stat()`, `git grep 'Stat(' | wc` gives 115 hits. I also think it should be done in a separate commit anyway.